compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -2617,9 +2588,18 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                             spans.push(trait_ref.path.span);
                         }
                         spans.push(self_ty.span);
-                        err.span_note(spans, &msg)
+                        err.subdiagnostic(NoteObligation::ImplDerivedObligationSpans {
+                            spans: MultiSpan::from_spans(spans),
+                            ty: parent_trait_pred.skip_binder().self_ty(),
+                            trait_path: parent_trait_pred
+                                .print_modifiers_and_trait_path()
+                                .to_string(),
+                        })
                     }
-                    _ => err.note(&msg),
+                    _ => err.subdiagnostic(NoteObligation::ImplDerivedObligation {
+                        ty: parent_trait_pred.skip_binder().self_ty(),
+                        trait_path: parent_trait_pred.print_modifiers_and_trait_path().to_string(),
```

Comment:
We might be able to implement `IntoDiagnosticArg` for `TraitPredPrintModifiersAndPath` too